packages/flutter/lib/src/material/dropdown.dart

```diff
@@ -1586,30 +1592,75 @@ class _DropdownButtonState<T> extends State<DropdownButton<T>> with WidgetsBindi
       },
     );
 
+    // When an InputDecoration is provided, use it instead of using an InkWell
```

Comment:
Yes we do. Splash effect on DropdownButton is not part of the spec and it was not there until https://github.com/flutter/flutter/pull/95906 added the InkWell.

If we want to keep the InkWell, I think the only solution to fix the issues listed in this PR description will be to add an option inside InputDecorator to add an optional InkWell (would be a non-breaking change but add a little more complexity to InputDecorator). Otherwise it is almost impossible to properly positioned and sized the InkWell has the InputDecorator container adjust on based on several parameters (border style, content padding, ect).

Reply:
I'm on board with removing the InkWell, thanks for explaining. I double checked the spec and you're right that it's not in there.